File: day1/day1.php
<?php

  echo "Advent of code: Day 1 - PHP\n";

  $floor = 0;
  $basementFound = false;
  $basementPosition = 0;

  $input = file_get_contents("input.txt");

  for($pos=0; $pos<= strlen($input); $pos++) {
    $move = substr($input, $pos, 1);
    if ( $move == '(' ) {
      $floor++;
    } else if ( $move == ')' ) {
      $floor--;
    }
    if ( $floor == -1 && !$basementFound ) {
      $basementFound = true;
      $basementPosition = $pos+1;
    }
  }

  print("Part One: Santa ended up on floor: " . $floor. "\n");
  print("Part Two: Santa entered the basement on position " . $basementPosition . "\n");
?>

File: day1/day1.go
package main

import (
  "io/ioutil"
)

func main() {

  println("Advent of code: Day 1 - Go")

  input, err := ioutil.ReadFile("./input.txt")
  if err != nil {
    panic(err)
  }

  floor := 0
  basementFound := false
  basementPosition := 0

  for pos := 0; pos <= len(input); pos++ {
    move := string(input[pos:pos+1])
    if move == "(" {
      floor += 1
    } else if move == ")" {
      floor -= 1
    }
    if floor == -1 && !basementFound {
      basementFound = true
      basementPosition = pos+1
    }
  }

  println("Part One: Santa ended up on floor:", floor)
  println("Part Two: Santa entered the basement on position", basementPosition)
}
